android-java/app/src/main/java/com/example/msaldemo/SingleAccountModeFragment.java

```diff
@@ -0,0 +1,383 @@
+package com.example.msaldemo;
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.microsoft.identity.client.AuthenticationCallback;
+import com.microsoft.identity.client.IAccount;
+import com.microsoft.identity.client.IAuthenticationResult;
+import com.microsoft.identity.client.IPublicClientApplication;
+import com.microsoft.identity.client.ISingleAccountPublicClientApplication;
+import com.microsoft.identity.client.PublicClientApplication;
+import com.microsoft.identity.client.SilentAuthenticationCallback;
+import com.microsoft.identity.client.exception.MsalClientException;
+import com.microsoft.identity.client.exception.MsalException;
+import com.microsoft.identity.client.exception.MsalServiceException;
+import com.microsoft.identity.client.exception.MsalUiRequiredException;
+
+import org.json.JSONObject;
+
+/**
+ * Implementation sample for 'Single account' mode.
+ * <p>
+ * If your app only supports one account being signed-in at a time, this is for you.
+ * This requires "account_mode" to be set as "SINGLE" in the configuration file.
+ * (Please see res/raw/auth_config_single_account.json for more info).
+ * <p>
+ * Please note that switching mode (between 'single' and 'multiple' might cause a loss of data.
+ */
+public class SingleAccountModeFragment extends Fragment {
+    private static final String TAG = SingleAccountModeFragment.class.getSimpleName();
+
+    /* UI & Debugging Variables */
+    Button signInButton;
+    Button signOutButton;
+    Button callGraphApiInteractiveButton;
+    Button callGraphApiSilentButton;
+    TextView scopeTextView;
+    TextView graphResourceTextView;
+    TextView logTextView;
+    TextView currentUserTextView;
+    TextView deviceModeTextView;
+
+    /* Azure AD Variables */
+    private ISingleAccountPublicClientApplication mSingleAccountApp;
+    private IAccount mAccount;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+                             ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        final View view = inflater.inflate(R.layout.fragment_single_account_mode, container, false);
+        initializeUI(view);
+
+        // Creates a PublicClientApplication object with res/raw/auth_config_single_account.json
+        PublicClientApplication.createSingleAccountPublicClientApplication(getContext(),
```

Comment:
Let's explicitly mention that this happens in the background, so the app developer is responsible for making sure that the PublicClientApplication object is initialized before attempting to use it.